cnf-tests/testsuites/e2esuite/dpdk/dpdk.go

```diff
@@ -1134,6 +1186,122 @@ func findNUMAForSRIOV(pod *corev1.Pod) (int, error) {
 	return -1, fmt.Errorf("failed to find the numa for pci %s", pciEnvVariableName)
 }
 
+func checkDownwardApi(pod *corev1.Pod, path string, c string) (string, error) {
+	var output string
+
+	switch {
+	case c == "label":
+		buff, err := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
+		if err != nil {
+			fmt.Println("Volume", path, "is not present")
+		}
+		output = buff.String()
+	case c == "IP":
+		buff, err := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
+		if err != nil {
+			fmt.Println("Volume", path, "is not present")
+		}
+		output = buff.String()
+	case c == "MAC":
+		buff, err := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
+		if err != nil {
+			fmt.Println("Volume", path, "is not present")
+		}
+		for _, line := range strings.Split(buff.String(), "\n") {
+			if strings.Contains(line, "mac_address") {
+				r := regexp.MustCompile(`[^\s"']+|"([^"]*)"|'([^']*)`)
+				arr := r.FindAllString(line, -1)
+
+				for _, v := range arr {
+					trim := v[:len(v)-1]
+					hw, _ := net.ParseMAC(trim)
+					if hw != nil {
+						output = fmt.Sprintln(hw)
+						break
+					}
+				}
+
+			}
+		}
+	case c == "PCI":
+		buff, err := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
+		if err != nil {
+			fmt.Println("Volume", path, "is not present")
+		}
+		for _, line := range strings.Split(buff.String(), "\n") {
+			if strings.Contains(line, "pci-address") {
+				r := regexp.MustCompile(`pci-address\\": \\".+?\\`)
+				arr := r.FindAllString(line, -1)
+				output = fmt.Sprintln(arr[0])
+				break
+			}
+		}
+	case c == "hugepages_request":
+		buff, _ := pods.ExecCommand(client.Client, *pod, []string{"cat", "/etc/podnetinfo/" + path})
```

Comment:
code for "hugepages_request" and "hugepages_limit" seems to be the same
fallthrough?

Reply:
I combined them into the same case with the and or operator ||